contrib/design-docs/quadlet_api/quadlet.yaml

```diff
@@ -0,0 +1,2955 @@
+
+    ## Target Audience
+
+    This API is designed for:
+    - System administrators managing containerized workloads
+    - DevOps engineers integrating containers with systemd
+    - Developers building container orchestration tools
+    - Platform engineers managing container infrastructure
+externalDocs:
+  description: Podman Quadlet Documentation
+  url: "https://docs.podman.io/en/latest/markdown/podman-quadlet.1.html"
+tags:
+  - name: libpod
+    description: Libpod quadlet operations.
+  - name: quadlet
+    description: Quadlet unit management operations.
+servers:
+  - url: "http://example-quadlet.com/v5.y.0/libpod/quadlets"
+  - url: "https://example-quadlet.com/v5.y.0/libpod/quadlets"
+security:
+  - main_auth: []
+paths:
+  "/":
+    get:
+      tags:
+        - libpod
+        - quadlet
+      summary: Get all quadlets
+      description: Retrieve a list of all quadlets.
+      operationId: getAllQuadletsLibpod
+      security:
+        - main_auth: []
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Quadlet"
+        "403":
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  "/{name}":
+    parameters:
+      - name: name
+        in: query
+        description: quadlet name to inspect and report
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - libpod
+        - quadlet
+      summary: Get quadlet by name
+      description: |
+        Read the quadlet system unit files by name and render them as JSON.
+      operationId: getQuadletByNameLibpod
+      parameters:
+        - $ref: "#/components/parameters/NameParam"
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Quadlet"
+        "403":
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "404":
+          description: Quadlet not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  "/{name}/build":
```

Comment:
Why are we doing separate endpoints for each unit type? I don't see a reason not to unify them